src/NodeApi/Native/JSNativeApi.Interop.cs

```diff
@@ -11,36 +12,200 @@ public static partial class JSNativeApi
 {
     // Node-API Interop definitions and functions.
     [SuppressUnmanagedCodeSecurity]
-    public static unsafe partial class Interop
+    public unsafe partial class Interop
     {
+        private readonly nint _libraryHandle;
+        private readonly nint[] _functions = new nint[(int)FunctionId.FunctionCount];
+
         private static bool s_initialized;
 
-        public static void Initialize()
+        public static Interop? Current { get; set; }
+
+        public Interop(nint libraryHandle) => _libraryHandle = libraryHandle;
+
+        public static void Initialize(nint libraryHandle)
         {
             if (s_initialized) return;
             s_initialized = true;
 
+            Current = new Interop(libraryHandle);
```

Comment:
Should it be a static class instead of singleton? I can't think of any reason it needs to be an instance.

Static method calls are slightly faster since there is one less level of indirection and no need for a null check.

Reply:
Thank you for the suggested change! I merged it and I also changed from `enum` fields to a `struct` to avoid extra bounds checking.